include/circt/Dialect/Calyx/CalyxPrimitives.td

```diff
@@ -99,4 +99,15 @@ def MemoryOp : CalyxPrimitive<"memory", [
       "ArrayRef<int64_t>":$addrSizes
     )>
   ];
+
+  let extraClassDeclaration = [{
+    size_t getNumAddresses() { return addrSizes().size(); }
```

Comment:
tiny nit: `getNumDimensions` ? I guess yours would be more in line with the op's names :)